### PR TITLE
Record views modified to display patient lists for tasks and product …

### PR DESCRIPTION
**Pull request by dczulada**
…tests

Button added to test executions to view patients

### CODE REVIEW

File: app/views/records/_records_list.html.erb [hunk 1]
```diff
@@ -31,7 +31,13 @@
...
       <% @records.each do |r| %>
         <tr>
-          <td><%= link_to full_name(r), record_path(:id => r.id), class: 'navbar-item' %></td>
+          <% if @task %>
+            <td><%= link_to full_name(r), record_path(:id => r.id, :task_id => @task.id), class: 'navbar-item' %></td>
+          <% elsif @product_test %>
+            <td><%= link_to full_name(r), record_path(:id => r.id, :product_test_id => @product_test.id), class: 'navbar-item' %></td>
+          <% else %>
+            <td><%= link_to full_name(r), record_path(:id => r.id), class: 'navbar-item' %></td>
+          <% end %>
```
Comment (thecristen) [inline app/views/records/_records_list.html.erb]: it looks like i'd left an erroneous `.navbar-item` class there... would you mind removing it

File: app/controllers/records_controller.rb [hunk 2]
```diff
@@ -35,15 +34,20 @@ def set_record_source
...
+      add_breadcrumb 'Master Patient List', :records_path
     elsif params[:product_test_id]
       @product_test = ProductTest.find(params[:product_test_id])
       @source = @product_test
+      add_breadcrumb 'Test: ' + @product_test.name, "/products/#{@product_test.product.id}/product_tests/#{@product_test.id}"
+      add_breadcrumb 'Patient List', records_path(product_test_id: @product_test.id)
     elsif params[:task_id]
       @task = Task.find(params[:task_id])
       @source = @task
+      add_breadcrumb 'Patient List', records_path(task_id: @task.id)
```
Comment (thecristen) [inline app/controllers/records_controller.rb]: it seems like it might be appropriate to provide some information to the user about the chosen task here?

**Comment by okeefm** [comment]
Similar to the other one, if you could provide screenshots of what it should look like, that would be awesome

File: app/controllers/records_controller.rb (hunk 2 above)
Comment (dczulada) [inline app/controllers/records_controller.rb]: I don't think there are any show pages for a task and all tasks share the same patient list as the product test.  Might make sense to remove the ability to filter patients on task.

**Comment by dczulada** [comment]
"View Patients" is a button under the Download Patients button 
screen shot 2016-01-28 at 1 24 59 pm

**Comment by dczulada** [comment]
The Patient List has a breadcrumbs back to the test
screen shot 2016-01-28 at 1 26 56 pm

File: app/controllers/records_controller.rb [hunk 3]
```diff
@@ -35,15 +34,16 @@ def set_record_source
...
       @bundle = Bundle.find(params[:bundle_id])
       @source = @bundle
+      add_breadcrumb 'Master Patient List', :records_path
     elsif params[:product_test_id]
       @product_test = ProductTest.find(params[:product_test_id])
       @source = @product_test
-    elsif params[:task_id]
-      @task = Task.find(params[:task_id])
-      @source = @task
+      add_breadcrumb 'Test: ' + @product_test.name, "/products/#{@product_test.product.id}/product_tests/#{@product_test.id}"
```
Comment (okeefm) [inline app/controllers/records_controller.rb]: Is there a reason this line kind of mixes string interpolation and string concatenation?

Comment (ssayer) [inline app/controllers/records_controller.rb]: Also we shouldn't be constructing URLs by hand, Use the URL/Path helpers generated by Rails

File: test/controllers/records_controller_test.rb [hunk 4]
```diff
@@ -31,14 +31,6 @@ class RecordsControllerTest < ActionController::TestCase
     assert assigns(:product_test)
   end
 
-  test 'should get index scoped to task' do
```
Comment (okeefm) [inline test/controllers/records_controller_test.rb]: Why did this get removed?

**Comment by okeefm** [comment]
One last note; otherwise looks good to me.

**Pull request merged**